modules/api/pkg/kubernetes/helper.go

```diff
@@ -37,18 +42,45 @@ func CheckContainerRuntime(ctx context.Context,
 	externalCluster *kubermaticv1.ExternalCluster,
 	externalClusterProvider provider.ExternalClusterProvider,
 ) (string, error) {
-	nodes, err := externalClusterProvider.ListNodes(ctx, masterClient, externalCluster)
+	clusterClient, err := externalClusterProvider.GetClient(ctx, masterClient, externalCluster)
+	if err != nil {
+		return "", err
+	}
+
+	controlPlaneNode, err := ListControlPlaneNode(ctx, clusterClient)
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch container runtime: not able to list control plane nodes %w", err)
+	}
+
+	for _, node := range controlPlaneNode.Items {
+		containerRuntimeVersion := node.Status.NodeInfo.ContainerRuntimeVersion
+		strSlice := strings.Split(containerRuntimeVersion, ":")
+		return strSlice[0], nil
```

Comment:
The "for" loop will iterate only the first slice element in that case. Is that intended?